pkg/rc/farm.go

```diff
@@ -138,24 +142,42 @@ START_LOOP:
 				}
 				foundChildren[rcField.ID] = struct{}{}
 
-				go func() {
+				go func(id fields.ID) {
+					defer func() {
+						if r := recover(); r != nil {
+							err := util.Errorf("Caught panic in rc farm: %s", r)
+							rcLogger.WithError(err).
+								WithField("rc_id", rcField.ID).
+								Errorln("Caught panic in rc farm")
+						}
+					}()
 					// disabled-ness is handled in watchdesires
 					for err := range newChild.WatchDesires(childQuit) {
 						rcLogger.WithError(err).Errorln("Got error in replication controller loop")
 					}
 
-					// NOTE: if WatchDesires experiences an unrecoverable error, we don't release the replication controller.
-					// However, it is unlikely that another farm instance would fare any better so that's okay
-				}()
+					// Release the child so that another farm can reattempt
+					rcf.childMu.Lock()
+					defer rcf.childMu.Unlock()
+					if _, ok := foundChildren[id]; !ok {
+						rcf.releaseChild(id)
```

Comment:
i'm realizing this is a mistake, I should be checking `rcf.children` for it not `foundChildren`. Of course it's in the latter, because we found it and attempted to operate on it.